v1/plugin/config_policy.go

```diff
@@ -20,122 +20,129 @@ limitations under the License.
 package plugin
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/rpc"
 )
 
 type ConfigPolicy struct {
-	IntegerRules map[string]integerRule
-	BoolRules    map[string]boolRule
-	StringRules  map[string]stringRule
-	FloatRules   map[string]floatRule
+	boolRules    map[string]*rpc.BoolPolicy
+	floatRules   map[string]*rpc.FloatPolicy
+	integerRules map[string]*rpc.IntegerPolicy
+	stringRules  map[string]*rpc.StringPolicy
 }
 
 func NewConfigPolicy() *ConfigPolicy {
 	return &ConfigPolicy{
-		IntegerRules: map[string]integerRule{},
-		BoolRules:    map[string]boolRule{},
-		StringRules:  map[string]stringRule{},
-		FloatRules:   map[string]floatRule{},
+		boolRules:    map[string]*rpc.BoolPolicy{},
+		floatRules:   map[string]*rpc.FloatPolicy{},
+		integerRules: map[string]*rpc.IntegerPolicy{},
+		stringRules:  map[string]*rpc.StringPolicy{},
 	}
 }
 
-// AddIntRule adds a given integerRule to the IntegerRules map.
-// This will overwrite any existing entry.
-func (c *ConfigPolicy) AddIntRule(key []string, in integerRule) {
-	k := strings.Join(key, ".") // Method used on daemon side in ctree
-	c.IntegerRules[k] = in
+// AddNewIntegerRule adds a new integerRule with the specified args to the integerRules map.
+// The required arguments are ns([]string), key(string), req(bool)
+// and optionally:
+//		config.SetDefaultInt(int64),
+//		config.SetMinInt(int64),
+//		config.SetMaxInt(int64),
```

Comment:
I think the comment on the NewIntRule was already wrong, but this is no longer in the config package. Could you update the examples to plugin.Set<Whatever>()? Applies to the others also.